crates/api/src/community/follow.rs

```diff
@@ -15,54 +16,61 @@ use lemmy_db_schema::{
-    let local_user_view = local_user_view_from_jwt(&data.auth, context).await?;
+  let community_id = data.community_id;
+  let community = Community::read(&mut context.pool(), community_id).await?;
+  let mut community_follower_form = CommunityFollowerForm {
+    community_id: data.community_id,
+    person_id: local_user_view.person.id,
+    pending: false,
+  };
 
-    let community_id = data.community_id;
-    let community = Community::read(&mut context.pool(), community_id).await?;
-    let mut community_follower_form = CommunityFollowerForm {
-      community_id: data.community_id,
-      person_id: local_user_view.person.id,
-      pending: false,
-    };
+  if data.follow {
+    if community.local {
+      check_community_ban(local_user_view.person.id, community_id, &mut context.pool()).await?;
+      check_community_deleted_or_removed(community_id, &mut context.pool()).await?;
 
-    if data.follow {
-      if community.local {
-        check_community_ban(local_user_view.person.id, community_id, &mut context.pool()).await?;
-        check_community_deleted_or_removed(community_id, &mut context.pool()).await?;
-
-        CommunityFollower::follow(&mut context.pool(), &community_follower_form)
-          .await
-          .with_lemmy_type(LemmyErrorType::CommunityFollowerAlreadyExists)?;
-      } else {
-        // Mark as pending, the actual federation activity is sent via `SendActivity` handler
-        community_follower_form.pending = true;
-        CommunityFollower::follow(&mut context.pool(), &community_follower_form)
-          .await
-          .with_lemmy_type(LemmyErrorType::CommunityFollowerAlreadyExists)?;
-      }
-    }
-    if !data.follow {
-      CommunityFollower::unfollow(&mut context.pool(), &community_follower_form)
+      CommunityFollower::follow(&mut context.pool(), &community_follower_form)
+        .await
+        .with_lemmy_type(LemmyErrorType::CommunityFollowerAlreadyExists)?;
+    } else {
+      // Mark as pending, the actual federation activity is sent via `SendActivity` handler
+      community_follower_form.pending = true;
+      CommunityFollower::follow(&mut context.pool(), &community_follower_form)
         .await
         .with_lemmy_type(LemmyErrorType::CommunityFollowerAlreadyExists)?;
     }
+  }
+  if !data.follow {
+    CommunityFollower::unfollow(&mut context.pool(), &community_follower_form)
+      .await
+      .with_lemmy_type(LemmyErrorType::CommunityFollowerAlreadyExists)?;
+  }
 
-    let community_id = data.community_id;
-    let person_id = local_user_view.person.id;
-    let community_view =
-      CommunityView::read(&mut context.pool(), community_id, Some(person_id), None).await?;
-    let discussion_languages = CommunityLanguage::read(&mut context.pool(), community_id).await?;
+  ActivityChannel::submit_activity(
+    SendActivityData::FollowCommunity(FollowCommunity {
+      community_id,
+      follow: data.follow,
+      auth: data.auth.clone(),
+    }),
```

Comment:
You can pass person and community directly so they dont have to be read from db again. See https://github.com/LemmyNet/lemmy/pull/3758

Reply:
Right, I made the change.